docs/Datadog.Trace/README.md

```diff
@@ -0,0 +1,229 @@
+
+```csharp
+using Datadog.Trace;
+
+var settings = TracerSettings.FromDefaultSources();
+
+// settings.AdoNetExcludedTypes.Add("MySql");      // <- Delete this line
+settings.Integrations["MySql"].Enabled = false; // <- Add this line
+```
+
+### `ElasticsearchNet5` integration ID has been removed
+
+In version 1.x, the integration ID for `Elasticsearch.Net` in version 5.x was `ElasticsearchNet5`, and for version 6.x+ was `ElasticsearchNet`. From version 2.0, support for the ID `ElasticsearchNet5` has been removed, and instead `ElasticsearchNet` now can be used for all versions of `Elasticsearch.Net`.
+
+Replace usages of `ElasticsearchNet5` with `ElasticsearchNet`
+
+```csharp
+using Datadog.Trace;
+
+var settings = TracerSettings.FromDefaultSources();
+
+settings.Integrations["ElasticsearchNet5"].Enabled = false; // <- Delete this line
+settings.Integrations["ElasticsearchNet"].Enabled = false;  // <- Add this line
+```
+
+### Obsolete APIs have been removed
+
+The following deprecated APIs have been removed.
+
+* `TracerSettings.DebugEnabled` has been removed. Set the `DD_TRACE_DEBUG` environment variable to `1` to enable debug mode.
+* `Tags.ForceDrop` and `Tags.ForceKeep` have been removed. Use `Tags.ManualDrop` and `Tags.ManualKeep` respectively instead.
+* `SpanTypes` associated with automatic instrumentation spans, such as `MongoDb` and `Redis` have been removed.
+* `Tags` associated with automatic instrumentation spans, such as `AmqpCommand` and `CosmosDbContainer`, have been removed.
+* `Tracer.Create()` has been removed. Use `Tracer.Configure()` instead.
+* `TracerSettings.AdoNetExcludedTypes` has been removed. Use `TracerSettings.Integrations` to disable automatic instrumentation for ADO.NET integrations.
+* Various internal APIs that were not intended for public consumption have been removed.
+
+In addition, some settings have been marked obsolete:
+
+* Environment variables `DD_TRACE_ANALYTICS_ENABLED`, `DD_TRACE_{0}_ANALYTICS_ENABLED`, and `DD_TRACE_{0}_ANALYTICS_SAMPLE_RATE` for controlling App Analytics are obsolete. App Analytics has been replaced with Tracing Without Limits. See [our documentation for details](https://docs.datadoghq.com/tracing/legacy_app_analytics/).
+* Setting `TracerSettings.AnalyticsEnabled` has similarly been marked obsolete.
+* Environment variable `DD_TRACE_LOG_PATH` is deprecated. Use `DD_TRACE_LOG_DIRECTORY` instead.
+
+### Introduction of ISpan and IScope
+
+In version 2.0 the public `Scope` and `Span` classes have been made internal. We now expose `IScope` and `ISpan` instead. The `Tracer` API has been updated accordingly. In most cases, this should not require any changes to your code, but if you are currently using explicit types (instead of inferred types), then you will need to replace usages of `Scope` with `IScope` and `Span` with `ISpan`:
```

Comment:
```suggestion
Version 2.0 makes the public `Scope` and `Span` classes internal. Instead, we now expose public `IScope` and `ISpan` interfaces and the `Tracer` API was updated accordingly. If you are currently using explicit types (instead of interring types with `var`), replace usages of `Scope` with `IScope` and `Span` with `ISpan`:
```